src/js/DropContainer.jsx

```diff
@@ -0,0 +1,281 @@
+               imageFile = URL.createObjectURL(evt);
+               // console.log('**DRAGGED** SELECTED FILE', evt);
+               this.setState({
+                  imageFile: imageFile
+               });
+            } else {
+               console.warn('Warning: no image selected');
+            }
+
+         // Setting the state image source
+         // With click button event
+         } else {
+            let evt = event.target.files[0];
+
+            // Image src to pass it to ImagePreview Component
+            if(evt !== undefined) {
+               imageFile = URL.createObjectURL(event.target.files[0]);
+               // console.log('SELECTED FILE', event.target.files[0]);
+
+               this.setState({
+                  imageFile: imageFile
+               });
+            } else {
+               console.warn('Warning: no image selected');
+            }
+         }
+      }
+   }
+
+   fileAccepted(file) {
+      return file.every(f => accepts(f, 'image/*'));
+   }
+
+   extractData(e) {
+      return e.dataTransfer && e.dataTransfer.items ? e.dataTransfer.items : [];
+   }
+   /**
+      * Handle the button footer color
+      * @param  {string} color
+      * @return {string}
+   */
+   setColor (val = 'blue') {
+      let color = val.toLowerCase();
+
+      if(val === 'green') {
+         color = 'green';
+      }else if (val === 'gray') {
+         color = 'gray';
+      } else if ( val === 'white') {
+         color = 'white';
+      }
+      return color;
+   }
+
+   /**
+      * Handle the button label text
+      * @param  {string} label
+      * @return {string}
+   */
+   setText (label = 'Upload') {
+      return label;
+   }
+
+   successBorder () {
+      this.setState({
+         borderStyle: 'successBorder'
+      });
+   }
+
+   warningBorder () {
+      this.setState({
+         borderStyle: 'warningBorder'
+      });
+   }
+
+   render () {
+      // define styles
+      let color = this.props.buttonColor,
+         text = this.props.buttonText,
+         borderStyle = this.state.borderStyle,
+         visibility = this.state.visibility,
+         outerClasses = classNames(css.groupWrapper, css[borderStyle]),
+         innerClasses = classNames(css[visibility]);
+      return (
+         <section
+            ref={(ref) => this.myTextInput = ref}
+            className={outerClasses}
+            onDragOver={this.onDragOver}
+            onDragLeave={this.onDragLeave}
+            onDragStart={this.onDragStart}
+            onDragEnter={this.onDragEnter}
+            onClick={this.onClick}
+            onDrop={this.onDrop}>
+            <section className={classNames(innerClasses, css.groupWrapper)}>
+               <ImagePreview
+                  imgSrc={this.state.imageFile}
+               />
+               <form className={classNames(css.flexContainer)}>
+                  <UploadButton
+                     ref='fileInput'
+                     accept='.png, .gif, .jpg, .jpeg'
```

Comment:
This shouldn't be hard coded and probably shouldn't be on a button anyway. From semantic issue, a button is just a clickable element. Probably should move the `input type='file'` out of the button and into the upload container.